sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -572,6 +572,14 @@ object SQLConf {
       "disable logging or -1 to apply no limit.")
     .createWithDefault(1000)
 
+  val WHOLESTAGE_MAX_LINES_PER_FUNCTION = buildConf("spark.sql.codegen.maxLinesPerFunction")
+    .internal()
+    .doc("The maximum lines of a single Java function generated by whole-stage codegen. " +
+      "When the generated function exceeds this threshold, " +
+      "the whole-stage codegen is deactivated for this subtree of the current query plan.")
+    .intConf
+    .createWithDefault(1500)
```

Comment:
When I modified it to 1600, the result is:
max function length of wholestagecodegen: Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
------------------------------------------------------------------------------------------------
codegen = F                                    467 /  507          1.4         712.7       1.0X
codegen = T maxLinesPerFunction = 1600        3191 / 3238          0.2        4868.7       0.1X
codegen = T maxLinesPerFunction = 1500         449 /  482          1.5         685.2       1.0X